### PR TITLE
add best practices for static versioning

### PR DESCRIPTION
**Pull request by scisco**
**Related Issue(s):** #
N/A

**Proposed Changes:**

1. Adds a section to best practices on how to achieve versioning in static catalogs

**PR Checklist:**

- [x] This PR has **no** breaking changes.
- [x] I have added my changes to the [CHANGELOG](https://github.com/radiantearth/stac-spec/blob/dev/CHANGELOG.md) **or** a CHANGELOG entry is not required.
- [x] API only: I have run `npm run generate-all` to update the [generated OpenAPI files](https://github.com/radiantearth/stac-spec/blob/dev/api-spec/README.md#openapi-definitions).

### CODE REVIEW

File: best-practices.md [hunk 1]
```diff
@@ -157,16 +157,16 @@ The main record should also provide a link to the versioned record following the
...
+--- root / collections / example_collection / items / my_item / versions / 01.json
 ```
 
 When `item1.json` is updated, the new `item1.json` includes a link to `versions/01.json` and is also copied to `versions/02.json`. This ensures that `02.json` includes a link to `01.json`
 
 ```
---- root / collections / example_collection / items / my_item / item1.json
---- root / collections / example_collection / items / my_item / item1.json / versions / 02.json
---- root / collections / example_collection / items / my_item / item1.json / versions / 01.json
+--- root / collections / example_collection / items / my_item / my_item.json
```
Comment (m-mohr) [inline best-practices.md]: Does this imply you have a folder per item?

Comment (matthewhanson) [reply best-practices.md]: Yeah, I think this is problematic, especially when relative paths are used in static catalogs. One might update the metadata, but point to the same assets (or most of the same assets).

I think all versions of the metadata should be in the same directory.

**Comment by m-mohr** [comment]
In https://github.com/radiantearth/stac-spec/pull/641/commits/da8efa40e4aac95da03247e632e4e9d096e48ab9 I added links between the Versioning Indicators Extension and the Best Practice Document.

File: best-practices.md [hunk 2]
```diff
@@ -131,6 +141,36 @@ Self-contained catalogs are not just for offline use, however - they are designe
...
 
+### Versioning for Catalogs
+
+The [Items and Collections API Version Extension](./api-spec/extensions/version) provides endpoints and semantics for keeping and accessing previous versions of Collections and Items. The same semantics can be used in static catalogs to preserve previous versions of the documents and link them together.
+
+In order to achieve this, the static catalog must make sure that for every record created, a copy of the record is also created in a separate location and it is named with the version id adopted by the catalog. See [here](/api-spec/extensions/version#version-id) for recommendations on versioning schema.
+
+The main record should also provide a link to the versioned record following the linking patterns described [here](/api-spec/extensions/version#link-rel-types). For every update to the record, the same cycle is repeated:
+
+1. add link from the updated record to the previous version
```
Comment (anayeaye) [inline best-practices.md]: Add

**Comment by matthewhanson** [comment]
@scisco Could you update this in response to comments above?  Would be good to get this into a 0.9.0 release.

**Comment by scisco** [comment]
@matthewhanson done

File: best-practices.md [hunk 3]
```diff
@@ -131,6 +141,36 @@ Self-contained catalogs are not just for offline use, however - they are designe
 on the cloud in object storage. They just aim to ease the burden of publishing, by not requiring lots of updating of links. 
 Adding a single `self` link at the root is recommended for online catalogs, turning it into a 'relative published catalog', as detailed below. This anchors it in an online location and enable provenance tracking.
 
+### Versioning for Catalogs
+
+The [Items and Collections API Version Extension](./api-spec/extensions/version) provides endpoints and semantics for keeping and accessing previous versions of Collections and Items. The same semantics can be used in static catalogs to preserve previous versions of the documents and link them together.
+
+In order to achieve this, the static catalog must make sure that for every record created, a copy of the record is also created in a separate location and it is named with the version id adopted by the catalog. See [here](/api-spec/extensions/version#version-id) for recommendations on versioning schema.
```
Comment (m-mohr) [inline best-practices.md]: The link should be `/api-spec/extensions/version/README.md#version-id`, I think. Otherwise it doesn't work on mobile.

**Pull request merged**